components/uniset/src/enum_props.rs

```diff
@@ -16,249 +18,304 @@ pub enum EnumeratedProperty {
     ScriptExtensions = 0x7000,
 }
 
+/// Enumerated Unicode general category types.
+/// GeneralSubcategory only supports specific subcategories (eg UppercaseLetter).
+/// It does not support grouped categories (eg Letter). For grouped categories, use GeneralCategory.
+#[derive(Copy, Clone, PartialEq, Debug, TryFromPrimitive, UnsafeFromPrimitive)]
+#[allow(missing_docs)] // TODO(#1030) - Add missing docs.
+#[repr(u8)]
+pub enum GeneralSubcategory {
```

Comment:
Keep in mind the tradeoffs being made by this `GeneralSubcategory` + `GeneralCategory` duo of enums. As I currently see them:

Pros:
* you have an enum that also behaves like a bitset with no extra deps
* this is probably fine in the return-a-UnicodeSet usage pattern (ex: regex)

Cons:
* for the get-property-value-at-this-code-point usage pattern, you still won't return any of the multi-category variants as the return value even if it's a `GeneralCategory` enum, so the code to test for "was that value in the Letter multi-category?" still may need to be a predicate function, but now you have a caveat that the set of valid values for `GeneralCategory` varies depending on the context.

Reply:
I'm not sure what you mean by "predicate function" here. We can write
```rust
fn is_char_in_general_category(c: char, gc: GeneralCategory) -> bool {
  let sc: GeneralSubcategory = get_subcategory(c);
  1 << (sc as u32) & gc as u32 != 0
}
```
In general, I think the rule of thumb is that we use GeneralCategory for inputs, and GeneralSubcategory for outputs.

Reply:
The "predicate function" could be

```rust
impl GeneralCategory {
    pub fn contains(sub: GeneralSubcategory) { ... }
}
```